Additional CELLRATE-related tweaks

atlantiscze opened the pull request:
- Technically a bugfix, converts two things that were using charge units directly to respect CELLRATE and use energy in watts.
- Mining drills now use CELLRATE correctly. Base power usage is 10kW, with advanced capacitor 5kW, and with super capacitor 3.3kW.
- Hardsuits should now use CELLRATE correctly.
- Tweaks power usage of hardsuit modules a bit, while i was at it. Many modules have received active power usage, which means they use power when running/on use. The power usages are now in joules/watts, instead of charge units. In the event of future cellrate changes, a similar problem should not occur.

Should be ready for review

atlantiscze committed:
Additional CELLRATE-related tweaks
- Mining drills now use CELLRATE correctly. Base power usage is 10kW, with advanced capacitor 5kW, and with super capacitor 3.3kW.
- Hardsuits should now use CELLRATE correctly.
- Tweaks power usage of hardsuit modules a bit, while i was at it. Many modules have received active power usage, which means they use power when running/on use. The power usages are now in joules/watts, instead of charge units. In the event of future cellrate changes, a similar problem should not occur.

Note: I am currently on a computer that simply isn't powerful enough to run the server to test this stuff. I am opening a PR for review, and to use Travis to compile this stuff, as my compilation time is about 30 minutes at least. I will finish this on Monday, most likely.

Techhead0 commented on code/modules/clothing/spacesuits/rig/rig.dm:
```diff
@@ -419,7 +419,7 @@
 	else if(!cell)
 		fail_msg = "<span class='warning'>There is no cell installed in the suit.</span>"
-	else if(cost && !cell.check_charge(cost * 10)) //TODO: Cellrate?
+	else if(cost && !cell.check_charge(cost * CELLRATE)) //TODO: Cellrate?
```
Begone, TODO comment!

Techhead0 commented on code/modules/mining/drilling/drill.dm:
```diff
@@ -265,8 +270,9 @@
-		cell.use(charge_use)
+	var/charge_to_use = actual_power_usage * CELLRATE
+	if(cell.charge >= charge_to_use)
+		cell.use(charge_to_use)
```
Cells have a proc for this. `checked_use()`
EDIT: In fact, this entire proc could be replaced with `return cell && cell.checked_use(actual_power_usage * CELLRATE)`

atlantiscze committed:
Additional adjustments
- IPC cooling units now use CELLRATE. On full load the power usage is 2kW. With default cell this means something above 30 minutes of battery time at least. This should be comparable to original battery time.
- Cyborgs and Drones no longer have insane-capacity power cells with manually set capacity in them. They get 1kWh cell (high capacity) by default now.
- Removes a TODO comment that was resolved.
- Additional tweak to mining drill code as suggested by Techhead.

atlantiscze committed:
Adds cellrate conversion to mechas
- Given the fact that mecha code is a HORRIBLE mess that urgently needs a refactor, it's possible i missed something. Though, exosuits should now respect CELLRATE.
- This includes movement, mining tools, and weapons. Weapons were balanced a bit. Energy weapons have medium energy cost when firing. Ballistic weapons, on the other hand, require large single-use spike when reloading (fabricating a new magazine).
- Exosuit chargers have been slightly refactored to respect CELLRATE. They will also no longer magically generate energy due to bad conversions now.

Techhead0 commented:
I did a quick search for more manually set cells:
Floodlights, mulebots, syndieborg, and that [dumb make-someone-a-borg proc](https://github.com/Baystation12/Baystation12/blob/dev/code/modules/mob/transform_procs.dm).

PsiOmegaDelta commented on code/game/mecha/mech_bay.dm:
```diff
@@ -6,9 +6,13 @@
 	density = 0
 	layer = TURF_LAYER + 0.1
 	anchored = 1
+	idle_power_usage = 200	// Some electronics, passive drain.
```
Why isn't this in KILOWATTS?
Repeat for every other defined power use without this or a similar constant.

atlantiscze replied:
Because it is in Watts. There is no constant for watts. KILOWATTS only does \* 1000 and MEGAWATTS only does \* 1000000

PsiOmegaDelta commented on code/modules/clothing/spacesuits/rig/modules/ninja.dm:
```diff
@@ -67,11 +67,11 @@
 	usable = 1
 	selectable = 1
-
+	module_cooldown = 60
```
Unmentioned balance change.

atlantiscze replied:
I'll be adding a changelog when i remove the DNM note from the PR.

atlantiscze committed:
Final tweaks, changelog
- Mulebots now use CELLRATE correctly and no longer have manually set power cell.
- Floodlights no longer use manually set power cell capacity
- Syndicate cyborg subtype no longer uses manually set power cell
- Transform procs no longer generate new power cell, this should be handled by cyborg's New() anyway.

atlantiscze commented:
The PR is ready for review. Changelog for balance-changing things added.
- Fixes #14045

PsiOmegaDelta commented:
The changelog is a good place to hide things you don't want to bring up in the OP I suppose.

:+1:
